lib/src/main/java/com/knuddels/jtokkit/TokenEncoderLarge.java

```diff
@@ -3,16 +3,20 @@
 
 import com.knuddels.jtokkit.api.IntArrayList;
 
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
 import java.util.Map.Entry;
 import java.util.TreeMap;
 
 import static com.knuddels.jtokkit.TokenEncoder.MAX_RANK;
+import static java.util.Objects.requireNonNull;
 
 final class TokenEncoderLarge {
     static int calculateTokensLarge(TokenEncoder tokenEncoder, int maxTokenCount, boolean keepEncodings, IntArrayList out, ByteArrayWrapper match, int length) {
         assert length > 1 : "Already filtered out";
 
-        TreeMap<Integer, TreeMap<Integer, RankNode>> rankMap = new TreeMap<>();
+        TreeMap<Integer, LinkedHashMap<Integer, RankNode>> rankMap = new TreeMap<>();
```

Comment:
`LinkedHashMap` is a lot faster than `TreeMap` and the found indices are already sorted so we might as well use it - seems to make this 2x faster - though here the speed doesn't matter that much, only the scaling. But the code is also simpler, so it's a win-win